### PR TITLE
[Basic] Fail when existing mkdir() target is not a directory

### PR DESCRIPTION
**Pull request by dmbryson**
rdar://problem/57807186

### CODE REVIEW

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

File: lib/BuildSystem/BuildSystem.cpp [hunk 1]
```diff
@@ -3130,27 +3130,27 @@ class SymlinkCommand : public Command {
...
       if (!parent.empty()) {
-        (void) bsci.getFileSystem().createDirectories(parent);
+        (void) fs.createDirectories(parent);
       }
     }
 
     // Create the symbolic link (note that despite the poorly chosen LLVM
     // name, this is a symlink).
-    //
-    // FIXME: Need to use the filesystem interfaces.
```
Comment (ddunbar) [inline lib/BuildSystem/BuildSystem.cpp]: Yay!!

File: products/llbuildSwift/BuildSystemBindings.swift [hunk 2]
```diff
@@ -581,6 +581,9 @@ public final class BuildSystem {
             _delegate.fs_get_file_info = { BuildSystem.toSystem($0!).fsGetFileInfo(String(cString: $1!), $2!) }
             // FIXME: This should be a separate callback, not shared with getFileInfo (or get FileInfo should take a parameter).
             _delegate.fs_get_link_info = { BuildSystem.toSystem($0!).fsGetFileInfo(String(cString: $1!), $2!) }
+
+            // FIXME: should support fs_create_symlink, but for now explicitly defers to built-in symlink
+            _delegate.fs_create_symlink = nil
```
Comment (ddunbar) [inline products/llbuildSwift/BuildSystemBindings.swift]: Any reason not to add this in now?

Comment (dmbryson) [reply products/llbuildSwift/BuildSystemBindings.swift]: Well, right now the FileSystem protocol in the swift bindings is pretty paltry.  It is fudging the getLinkInfo call by using getFileInfo (🤢).  The comment on BuildSystemBindings.swift:368 mentions a desire to sit on top of the protocol from SwiftPM.   Probably makes sense just clean the whole thing up when we can sit on TSC.

Comment (ddunbar) [reply products/llbuildSwift/BuildSystemBindings.swift]: OK!

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Pull request merged**